src/components/Table/GrantPeriod/GrantPeriodTable.tsx

```diff
@@ -0,0 +1,204 @@
+import { useApolloClient, useLazyQuery } from "@apollo/client";
+import {
+	Box,
+	createStyles,
+	IconButton,
+	makeStyles,
+	MenuItem,
+	Paper,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableHead,
+	TableRow,
+	Theme,
+	Typography,
+} from "@material-ui/core";
+import MoreVertOutlinedIcon from "@material-ui/icons/MoreVertOutlined";
+import React, { useEffect, useState } from "react";
+
+import { useDashBoardData } from "../../../contexts/dashboardContext";
+import { FETCH_GRANT_PERIODS } from "../../../graphql/grantPeriod/query";
+import { FORM_ACTIONS } from "../../../models/constants";
+import { IGrantPeriod } from "../../../models/grantPeriod/grantPeriodForm";
+import GrantPeriodDialog from "../../GrantPeriod/GrantPeriod";
+import SimpleMenu from "../../Menu/Menu";
+import TableSkeleton from "../../Skeletons/TableSkeleton";
+
+const useStyles = makeStyles({
+	table: {
+		minWidth: 650,
+	},
+});
+const StyledTableHeader = makeStyles((theme: Theme) =>
+	createStyles({
+		th: { color: theme.palette.primary.main },
+		tbody: {
+			"& tr:nth-child(even) td": { background: "#F5F6FA" },
+			"& td.MuiTableCell-root": {
+				paddingTop: "1px",
+				paddingBottom: "1px",
+			},
+		},
+	})
+);
+
+function resolveJSON(obj: { [key: string]: any }, keys: string) {
```

Comment:
functions like these should be in a `util` file

Reply:
done